### PR TITLE
Add continuous_update keyword for option widgets and fix a bunch of ipywidget 7 related bugs.

### PR DESCRIPTION
**Pull request by korsbo**
This minor update adds the kwarg continuous_update=Bool (true by default) to the option widgets. 

My motivation for this was to allow non-continuous updates of selection_sliders.

### CODE REVIEW

**Comment by korsbo** [comment]
I added two more commits which fixes layout problem with ipywidgets 7+.

1. sliders and selection_sliders could no longer be displayed vertically #202. 
2. vbox() behaved like hbox #198. 

I must however admit that I have not tested these changes on a system with ipywidgets < 7 for backwards compatibilty.

File: src/widgets.jl [hunk 1]
```diff
@@ -362,7 +364,7 @@ function Options(view::Symbol,
 end
 
 function getoptions(options)
-    opts = OrderedDict()
+    opts = OrderedDict{String, eltype(options)}()
```
Comment (JobJob) [inline src/widgets.jl]: Are these keys always Strings? Curious: why was this needed?

**Comment by korsbo** [comment]
I'm sorry that my correspondence has been slow, it has been a hectic week. 

I'm having trouble testing for compatibility with older ipywidgets. I started working on these functionalities for ipywidgets 7+ precisely because Interact was not displaying any widgets if I used ipywidgets 6. So when I now revert to 6.0.0 for testing the widgets does not show up, but this is likely due to some other error.

File: src/widgets.jl (hunk 1 above)
Comment (korsbo) [reply src/widgets.jl]: Yes, the keys are always strings, all the elements are added by either
```julia
addoption!(opts, v::Union{Pair, Tuple{Any, Any}}) = opts[string(v[1])] = v[2]
```
or
```julia
addoption!(opts, v) = opts[string(v)] = v
```

Changing this line allows for increased type-specificity of signal( selection( input_array; multi=true))). Before, this signal always returned an Array{Any, 1}. Now it will instead return an array with the same type as the input_array. 

This enables a better compile-time optimisation of any function which takes this as an input.

Comment (korsbo) [reply src/widgets.jl]: `opts = OrderedDict{Any, eltype(options)}() ` would also have worked. The type of the keys are not important, but the type of the value-container is.

Comment (korsbo) [reply src/widgets.jl]: Also, the selection_slider has functionality for syncing to the nearest value if it does not correspond exactly to one of the possible values. To access this functionality, one must pass a 
```julia
syncselnearest = view == :SelectionSlider && typ <: Real && syncnearest

if syncnearest
```
statement. `typ` is defined by `typ=valtype(options.dict)`, which previously always were `Any`. Thus the selection_slider's syncnearest functionality was not possible to reach (`Any` is not a subtype of `Real`).

**Comment by korsbo** [comment]
I just noticed that my fix for the initial values of selection_slider works for some values but not others. I currently do not know why, but I'll have a look.

File: src/widgets.jl (hunk 1 above)
Comment (korsbo) [reply src/widgets.jl]: Also, the selection_slider has functionality for syncing to the nearest value if it does not correspond exactly to one of the possible values. To access this functionality, one must pass a 
```julia
syncselnearest = view == :SelectionSlider && typ <: Real && syncnearest

if syncnearest
```
statement. `typ` is defined by `typ=valtype(options.dict)`, which previously always were `Any`. Thus the selection_slider's syncnearest functionality was not possible to reach (`Any` is not a subtype of `Real`).

**Comment by rafaqz** [comment]
Any news on this?

Neither 6.0 or 7.0 widgets are functional for me...

**Comment by korsbo** [comment]
I cloud help troubleshooting. This PR works fine for me on ipywidgests 7.1.0. What is the problem for you?

**Comment by rafaqz** [comment]
The big red "all checks failed" banner scared me off.  But I tried it. Running Pkg.build("Interact") gives this warning currently:
                                                                                                                 
LoadError: UndefVarError: @PYTHON_str not defined                                                                 
while loading /home/raf/.julia/v0.6/Interact/deps/build.jl, in expression starting on line 3

But otherwise it works! Thanks.

**Comment by korsbo** [comment]
I have been unable to verify that this PR does not break Interact.jl for people on ipywidgets 6 (since I could not get that to work even before this PR). Would you be able to do this? 

run 
```bash
pip install ipywidgets==6.0.0
```

and see if Interact still works after that?

**Comment by JobJob** [comment]
You'll IMO need to also put `5.0.0` in the file `Pkg.dir()/Interact/deps/ipywidgets_version`, if that doesn't work try `6.0.0`, sorry this is still a bit of a shambles.

**Comment by rafaqz** [comment]
6.0 widgets don't show up at all for me with or without this pull request, using either my distros packages or the IJulia Condo setup with 6.0 swapped in for 7.1. They recently stopped working for some reason.

So I cant test this for 6.0 unfortunately. But I do know that Pkg.build("Interact") has the same error on 6.0 as 7.1, and that would need to be fixed at least.

**Comment by rafaqz** [comment]
Can we depreciate ipywidgets 6.0? It seems that most people are migrating anyway, and we really need to have at least one fully working version. At the moment both options seem to be very broken.

**Comment by JobJob** [comment]
> @PYTHON_str not defined
Have you tried master? This PR doesn't have changes in #206 that fixed #205 yet

**Comment by rafaqz** [comment]
Master doesn't have the @Python error, its just broken on this branch. I guess the merge will fix that!

**Comment by korsbo** [comment]
I can merge master into the PR so that we can ensure that it works.

**Comment by JobJob** [comment]
please do - or rebase off master

**Comment by korsbo** [comment]
crap, that was not right. I lost a few commits. I'll fix shortly.

**Comment by JobJob** [comment]
Looks ok to me are you sure?

**Comment by korsbo** [comment]
I'm trying to figure that out.  After the rebase, I did a forced push to my fork. I then realised that I have two clones on my machine (stupid) and that the one I pushed from is two commits behind. I did a new forced push from the up-to-date clone and now I think that everything is ok.

**Comment by JobJob** [comment]
ummm looks like it was ok, but now you re-pushed the first 2 commits which are now conflicts. I think if you reset to dacbe86fb1fc4c8ddb5b5ed403f74fbe452a7acb and re-force push you should be good.

**Comment by JobJob** [comment]
see: https://github.com/JuliaGizmos/Interact.jl/pull/201/commits/243cbc4aa616884962e123779357873879271de0

**Comment by JobJob** [comment]
and https://github.com/JuliaGizmos/Interact.jl/pull/201/commits/4fe80f1f0806e9564fcf671db5e3a97c54d368a7

**Comment by korsbo** [comment]
Aah, I see. I'll fix.

**Comment by JobJob** [comment]
I would: `git reset --hard dacbe86fb1fc4c8ddb5b5ed403f74fbe452a7acb`

**Comment by JobJob** [comment]
then `git push -f`

**Comment by korsbo** [comment]
done.

**Comment by JobJob** [comment]
Ok looks good - I'm just trying it out with different ipywidgets/widgetnbextension versions now

**Comment by korsbo** [comment]
Great, I have been unable to get any version of Interact working with ipywidgets < 7. I'm therefore happy that someone else do the checks for backwards compatibility.

**Comment by JobJob** [comment]
All seems to work fine. Thanks so much for this. Sorry it took so long to get it done!

**Comment by korsbo** [comment]
Cheers!
